docs/openid.md

```diff
@@ -89,7 +90,49 @@ oidc_config:
        localpart_template: "{{ user.name }}"
        display_name_template: "{{ user.name|capitalize }}"
 ```
+### [Keycloak][keycloak-idp]
 
+[Keycloak][keycloak-idp] is an opensource IDP maintained by Red Hat. 
+
+Follow the [Getting Started Guide](https://www.keycloak.org/getting-started/) to install Keycloak and set up a realm.
+
+1. `Clients -> Create`
+
+2. Fill Fields
+
+| Field | Value |
+|-----------|-----------|
+| Client ID | `synapse` |
+| Client Protocol | `openid-connect` |
+
+3. `Save`
+
+4. Fill Fields
+
+| Field | Value |
+|-----------|-----------|
+| Client ID | `synapse` |
+| Enabled | `On` |
+| Client Protocol | `openid-connect` |
+| Access Type | `confidential` |
+| Valid Redirect URIs | `matrix.domain.tld/_synapse/oidc/callback` |
```

Comment:
In the other examples it seems we've used `[synapse public baseurl]`:

```suggestion
| Valid Redirect URIs | `[synapse public baseurl]/_synapse/oidc/callback` |
```

Reply:
https://github.com/matrix-org/synapse/blame/0637994ec8a457b5d59570ccc594135c0cfcf1bf/docs/openid.md#L118